### PR TITLE
Add unit test for #4423

### PR DESCRIPTION
**Pull request by cursedcoder**

### CODE REVIEW

File: test/core/Application.js [hunk 1]
```diff
@@ -48,4 +48,44 @@ describe('PIXI.Application', function ()
...
+        {
+            const filter = new PIXI.filters.AlphaFilter();
+
+            sprite.filters = [filter];
+
+            iterations--;
+
+            if (iterations !== 0)
+            {
+                setTimeout(() =>
```
Comment (bigtimebuddy) [inline test/core/Application.js]: What about `setTimeout(iterate, 200);` instead?

File: test/core/Application.js [hunk 2]
```diff
@@ -48,4 +48,44 @@ describe('PIXI.Application', function ()
...
+    it('should be able to destroy filter manager gracefully', function (done)
+    {
+        const app = new PIXI.Application();
+
+        document.body.appendChild(app.view);
+
+        const sprite = new PIXI.Sprite(PIXI.Texture.WHITE);
+
+        sprite.position.x = 60;
+        sprite.position.y = 60;
```
Comment (bigtimebuddy) [inline test/core/Application.js]: Is setting the position necessary?

Comment (cursedcoder) [reply test/core/Application.js]: It is a test, why not.

File: test/core/Application.js [hunk 3]
```diff
@@ -48,4 +48,44 @@ describe('PIXI.Application', function ()
...
+
+        app.stage.addChild(sprite);
+
+        let iterations = 3;
```
Comment (bigtimebuddy) [inline test/core/Application.js]: Does this error reproduce with less than 3 iterations of applying a filter? Do they need to be applied asynchronously? What about?
```js
sprite.filters = [filter];
app.render();
sprite.filters = [filter];
app.render();
sprite.filters = [filter];
app.render();
```

Comment (cursedcoder) [reply test/core/Application.js]: I've tried to use `ticker` – but no errors.

**Comment by cursedcoder** [comment]
@bigtimebuddy dunno, `9fdde4e` was enough to reproduce a red test on my local machine, I wonder what's happening on travis side.

**Comment by bigtimebuddy** [comment]
WebGL based rendering does not run on Travis. So this would reproduce failed locally but not on Travis.

**Comment by cursedcoder** [comment]
@bigtimebuddy if you are OK with PR, want me to reorganize commits?

File: src/core/renderers/webgl/managers/FilterManager.js [hunk 4]
```diff
@@ -233,9 +233,10 @@ export default class FilterManager extends WebGLManager
             else
             {
                 shader = filter.glShaders[renderer.CONTEXT_UID] = new Shader(this.gl, filter.vertexSrc, filter.fragmentSrc);
-                this.managedFilters.push(filter);
             }
 
+            this.managedFilters.push(filter);
```
Comment (bigtimebuddy) [inline src/core/renderers/webgl/managers/FilterManager.js]: This is a mistake right?

**Comment by bigtimebuddy** [comment]
This is better, don’t worry about the commit history. I squash them anyways when merging.

**Comment by cursedcoder** [comment]
@bigtimebuddy ok, then we are ready to go.

File: src/core/renderers/webgl/managers/FilterManager.js (hunk 4 above)
Comment (cursedcoder) [reply src/core/renderers/webgl/managers/FilterManager.js]: idea was to reproduce error on travis

**Comment by bigtimebuddy** [comment]
Thanks @cursedcoder! Sorry for the merge delay.

**Pull request merged**